src/transport/smtp/async_pool.rs

```diff
@@ -0,0 +1,100 @@
+use std::time::Duration;
+
+use crate::transport::smtp::{
+    async_transport::AsyncSmtpClient, client::AsyncSmtpConnection, error::Error, Tokio1Connector,
+};
+
+use async_trait::async_trait;
+use bb8::{ManageConnection, Pool, PooledConnection};
+
+#[derive(Debug, Clone)]
+#[allow(missing_copy_implementations)]
+#[cfg_attr(docsrs, doc(cfg(feature = "tokio1")))]
+pub struct AsyncPoolConfig {
+    min_idle: u32,
+    max_size: u32,
+    connection_timeout: Duration,
+    idle_timeout: Duration,
+}
+
+impl Default for AsyncPoolConfig {
+    fn default() -> Self {
+        Self {
+            min_idle: 0,
+            max_size: 10,
+            connection_timeout: Duration::from_secs(30),
+            idle_timeout: Duration::from_secs(60),
+        }
+    }
+}
+
+impl AsyncPoolConfig {
+    /// Create a new async pool configuration with default values
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Minimum number of idle connections
+    ///
+    /// Defaults to `0`
+    pub fn min_idle(mut self, min_idle: u32) -> Self {
+        self.min_idle = min_idle;
+        self
+    }
+
+    /// Maximum number of pooled connections
+    ///
+    /// Defaults to `10`
+    pub fn max_size(mut self, max_size: u32) -> Self {
+        self.max_size = max_size;
+        self
+    }
+
+    /// Connection timeout
+    ///
+    /// Defaults to `30 seconds`
+    pub fn connection_timeout(mut self, connection_timeout: Duration) -> Self {
+        self.connection_timeout = connection_timeout;
+        self
+    }
+
+    /// Connection idle timeout
+    ///
+    /// Defaults to `60 seconds`
+    pub fn idle_timeout(mut self, idle_timeout: Duration) -> Self {
+        self.idle_timeout = idle_timeout;
+        self
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn build<C: ManageConnection>(&self, client: C) -> Pool<C> {
+        Pool::builder()
+            .min_idle(Some(self.min_idle))
+            .max_size(self.max_size)
+            .connection_timeout(self.connection_timeout)
+            .idle_timeout(Some(self.idle_timeout))
+            .build_unchecked(client)
+    }
+}
+
+#[async_trait]
+impl ManageConnection for AsyncSmtpClient<Tokio1Connector> {
+    type Connection = AsyncSmtpConnection;
+    type Error = Error;
+
+    async fn connect(&self) -> Result<Self::Connection, Self::Error> {
+        let conn = self.connection().await?;
+        Ok(conn)
+    }
+
+    async fn is_valid(&self, conn: &mut PooledConnection<'_, Self>) -> Result<(), Self::Error> {
+        if conn.test_connected().await {
+            return Ok(());
+        }
+        Err(Error::Client("is not connected anymore"))
+    }
+
+    fn has_broken(&self, conn: &mut Self::Connection) -> bool {
+        conn.has_broken()
+    }
```

Comment:
bb8 doesn't seem to have a method that gets called before the connection is dropped, and we don't have async drop yet. How do be call `conn.quit()` :thinking:? I'll look into it. @amousset probably knows better about what happens with connections closing without the client sending the quit command.

Reply:
That's a very good point. Indeed it seems that bb8 doesn't implement such method. I can think of two scenarios for when `bb8` drops the connection.  
1. When the connection is broken, i.e. `has_broken` returns `true` or when `is_valid` returns an error. In my opinion, calling `quit` here would be redundant since the connection is no longer valid in any case.
2. When the whole connection pool is dropped. In this case I believe that the underlying tcp connection (in `async_net.rs`) is implicitly dropped which would close the connection to the SMTP server in any case. I agree that it would be better to explicitly call `quit` when disconnecting in order to be RFC compliant.

I think this issue is also relevant for the sync pool implementation with `r2d2`. 
I would be happy to make any changes to the PR that you recommend!

Reply:
I guess there isn't an 100% good solution for this. `quit` could be called on `Drop` in the case of sync connections, but for the async version even spawning the `quit` `Future` onto the executor probably won't guarantee that it'll get polled to completion in case the executor is shutting down.

Reply:
I guess we can mark this as resolved for now based on the follow up [comment](https://github.com/lettre/lettre/pull/550#issuecomment-786899603)!